internal/ethapi/api.go

```diff
@@ -1516,10 +1521,20 @@ func (api *TransactionAPI) SendTransaction(ctx context.Context, args Transaction
 // on a given unsigned transaction, and returns it to the caller for further
 // processing (signing + broadcast).
 func (api *TransactionAPI) FillTransaction(ctx context.Context, args TransactionArgs) (*SignTransactionResult, error) {
-	args.blobSidecarAllowed = true
-
 	// Set some sanity defaults and terminate on failure
-	if err := args.setDefaults(ctx, api.b, false); err != nil {
+	config := setDefaultConfig{
+		skipGasEstimation:  false,
+		blobSidecarAllowed: true,
+		blobSidecarVersion: types.BlobSidecarVersion0,
+	}
+	if len(args.Blobs) > 0 {
+		chainHead := api.b.CurrentHeader()
+		isOsaka := api.b.ChainConfig().IsOsaka(chainHead.Number, chainHead.Time)
+		if isOsaka {
+			config.blobSidecarVersion = types.BlobSidecarVersion1
```

Comment:
Note, the legacy proofs submitted by users will be implicitly converted to cell proofs.